infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServerCache.java

```diff
@@ -126,28 +73,40 @@ public void delete(@Nonnull ConflictTxStream key, Long globalAddress, @Nonnull R
      * @return global address
      */
     public Long getIfPresent(ConflictTxStream conflictKey) {
-        return conflictCache.getIfPresent(conflictKey);
+        return cacheHash.get(conflictKey);
+    }
+
+    private long firstAddress() {
+        if (cacheEntries.peek() == null)
```

Comment:
Please replace by `if(!cacheEntries.isEmpty())`

Reply:
This seems to be incorrect. If the priority queue fills up with [1, 2, 3, 4, 5], then `invalidateUpTo` is called to 6, then it fills up with [6, 7, 8, 9], `firstAddress` will return `Address.NOT_FOUND`. Then a put cannot cause an eviction because 6 is greater than `firstAddress()` + 1. Essentially, the window cannot slide.